### PR TITLE
The Milestone Commit

### PR DESCRIPTION
**Pull request by mctaylors**
mctaylors:
- updated readme 7 times (and only adding new logo from /about)
- [removed](https://github.com/TeamOctolings/Boyfriend/pull/48/commits/aeeb3d4399c31df70b47ccdf59f6963fdb68e9ec) bot footer from created event embed on the second try
- [changed](https://github.com/TeamOctolings/Boyfriend/pull/48/commits/4b9b91d9e4d2289d9aad4e600f5ca6a424638a6e) cdn from discord to upload.systems

Octol1ttle:
- Guild settings code has been overhauled. Instead of instances of a `GuildConfiguration` class being (de-)serialized when used with listing and setting options provided by reflection, there are now multiple `Option` classes responsible for the type of option they are storing. The classes support getting a value, validating and setting values with Results, and getting a user-friendly representation of these values. This makes use of polymorphism, providing clean and easier to use and refactor code.
- Gateway event responders have been split into their own separate files, which should make it easier to find and modify responders when needed.
- Warning suppressions regarding unused and never instantiated classes have been replaced by `[ImplicitUse]` annotations provided by `JetBrains.Annotations`. This avoids hiding real issues and provides a better way to suppress false warnings while being explicit.
- It is no longer possible to execute some slash commands if they are run without the correct permissions
- Dependencies are now more explicitly defined

neroduckale:
 - Made easter eggs case-insensitive

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <p align="center">
-<img src="https://github.com/TeamOctolings/Boyfriend/docs/assets/boyfriend-dark.png#gh-light-mode-only" 
+<img src="https://github.com/TeamOctolings/Boyfriend/blob/milestone-commit/docs/assets/boyfriend-dark.png#gh-light-mode-only" 
     alt="Boyfriend logo" width="75%"/>
-<img src="https://github.com/TeamOctolings/Boyfriend/docs/assets/boyfriend-light.png#gh-dark-mode-only" 
+<img src="https://github.com/TeamOctolings/Boyfriend/blob/milestone-commit/docs/assets/boyfriend-light.png#gh-dark-mode-only"
```
Comment (Octol1ttle) [inline docs/README.md]: This hardcodes the logo images to this branch of this repository of this organization. Not ideal.

File: docs/assets/boyfriend-dark.png
Comment (Octol1ttle) [inline docs/assets/boyfriend-dark.png:1]: Why have the logos been turned to their oversimplified versions?

**Review by Octol1ttle — CHANGES_REQUESTED** [review]

File: docs/assets/boyfriend-dark.png
Comment (mctaylors) [reply docs/assets/boyfriend-dark.png:1]: because yes

File: docs/README.md (hunk 1 above)
Comment (mctaylors) [reply docs/README.md]: i'll fix that tomorrow

Comment (mctaylors) [reply docs/README.md]: ok maybe on monday (i love my internet providers)

File: src/Responders/MessageReceivedResponder.cs [hunk 2]
```diff
@@ -24,8 +24,7 @@ public class MessageCreateResponder : IResponder<IMessageCreate> {
                 "whoami"  => "`nobody`",
                 "сука !!" => "`root`",
                 "воооо"   => "`removing /...`",
-                "пон" =>
-                    "https://cdn.discordapp.com/attachments/837385840946053181/1087236080950055023/vUORS10xPaY-1.jpg",
+                "пон" => "https://cdn.upload.systems/uploads/2LNfUSwM.jpg",
```
Comment (Octol1ttle) [inline src/Responders/MessageReceivedResponder.cs:27]: https://github.com/TeamOctolings/Boyfriend/blob/milestone-commit/src/Commands/AboutCommandGroup.cs#L70

File: src/Responders/MessageReceivedResponder.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@ public class MessageCreateResponder : IResponder<IMessageCreate> {
 
     public Task<Result> RespondAsync(IMessageCreate gatewayEvent, CancellationToken ct = default) {
         _ = _channelApi.CreateMessageAsync(
-            gatewayEvent.ChannelID, ct: ct, content: gatewayEvent.Content switch {
+            gatewayEvent.ChannelID, ct: ct, content: gatewayEvent.Content.ToLower() switch {
```
Comment (Octol1ttle) [inline src/Responders/MessageReceivedResponder.cs:23]: Please use `.ToLowerInvariant()`

**Review by Octol1ttle — CHANGES_REQUESTED** [review]

**Review by Octol1ttle — DISMISSED** [review]

**Review by Octol1ttle — APPROVED** [review]

**Pull request merged**